src/mongocxx/include/mongocxx/v1/change_stream.hpp

```diff
@@ -95,7 +95,9 @@ class change_stream {
     ///
     /// @important All iterators associated with the same change stream object share the same state.
     ///
-    /// This function advances the underlying cursor to obtain the next (first) event document.
+    /// This function advances the underlying cursor to obtain the first available event document. The underlying cursor
+    /// is only advanced at most once: consecutive calls to `this->begin()` do not advance the underlying cursor state.
+    /// To obtain subsequent available event documents, the resulting iterator must be incremented instead.
     ///
     /// @throws mongocxx::v1::server_error when a server-side error is encountered.
```

Comment:
Suggest adding similar warning as iterator `operator++` since this may advance to the next event document:

```suggestion
    /// @warning Invalidates all views to the current event document and resume token.
    ///
    /// @throws mongocxx::v1::server_error when a server-side error is encountered.
```